## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 441147`
Commit: `a2e2c76d266fe0c18890461604768d81bd19bcbb`

---
### `x86_64-linux`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_10.tuxedo-drivers</li>
    <li>linuxKernel.packages.linux_5_15.tuxedo-drivers</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 11 packages built:</summary>
  <ul>
    <li>linuxKernel.packages.linux_6_1.tuxedo-drivers</li>
    <li>linuxPackages.tuxedo-drivers (linuxKernel.packages.linux_6_12.tuxedo-drivers)</li>
    <li>linuxKernel.packages.linux_hardened.tuxedo-drivers (linuxKernel.packages.linux_6_12_hardened.tuxedo-drivers)</li>
    <li>linuxPackages_latest.tuxedo-drivers (linuxKernel.packages.linux_6_16.tuxedo-drivers)</li>
    <li>linuxKernel.packages.linux_6_6.tuxedo-drivers</li>
    <li>linuxPackages_latest-libre.tuxedo-drivers (linuxKernel.packages.linux_latest_libre.tuxedo-drivers)</li>
    <li>linuxPackages-libre.tuxedo-drivers (linuxKernel.packages.linux_libre.tuxedo-drivers)</li>
    <li>linuxPackages_lqx.tuxedo-drivers (linuxKernel.packages.linux_lqx.tuxedo-drivers)</li>
    <li>linuxPackages_xanmod.tuxedo-drivers (linuxKernel.packages.linux_xanmod.tuxedo-drivers)</li>
    <li>linuxPackages_xanmod_latest.tuxedo-drivers (linuxKernel.packages.linux_xanmod_latest.tuxedo-drivers, linuxPackages_xanmod_stable.tuxedo-drivers)</li>
    <li>linuxPackages_zen.tuxedo-drivers (linuxKernel.packages.linux_zen.tuxedo-drivers)</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>linuxKernel.packages.linux_5_10.tuxedo-drivers</summary>
<pre>/build/source/src/tuxedo_nb05/tuxedo_nb05_power_profiles.c: In function &#x27;tuxedo_nb05_power_profiles_probe&#x27;:
/build/source/src/tuxedo_nb05/tuxedo_nb05_power_profiles.c:331:9: warning: ISO C90 forbids mixed declarations and code [8;;https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wdeclaration-after-statement-Wdeclaration-after-statement8;;]
  331 |         const struct platform_device_info pinfo = {
      |         ^~~~~
/build/source/src/tuxedo_nb05/tuxedo_nb05_power_profiles.c: In function &#x27;tuxedo_nb05_power_profiles_remove&#x27;:
/build/source/src/tuxedo_nb05/tuxedo_nb05_power_profiles.c:360:9: error: implicit declaration of function &#x27;timer_delete&#x27;; did you mean &#x27;timer_reduce&#x27;? [8;;https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wimplicit-function-declaration-Werror=implicit-function-declaration8;;]
  360 |         timer_delete(&amp;profile_changed_timer);
      |         ^~~~~~~~~~~~
      |         timer_reduce
/build/source/src/tuxedo_nb05/tuxedo_nb05_power_profiles.c:361:9: warning: ISO C90 forbids mixed declarations and code [8;;https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wdeclaration-after-statement-Wdeclaration-after-statement8;;]
  361 |         struct driver_data_t *driver_data = dev_get_drvdata(&amp;wdev-&gt;dev);
      |         ^~~~~~
cc1: some warnings being treated as errors
make[5]: *** [/nix/store/qsfd9ca611zxw577iyabjchmdn845b79-linux-5.10.242-dev/lib/modules/5.10.242/source/scripts/Makefile.build:286: /build/source/src/tuxedo_nb05/tuxedo_nb05_power_profiles.o] Error 1
make[4]: *** [/nix/store/qsfd9ca611zxw577iyabjchmdn845b79-linux-5.10.242-dev/lib/modules/5.10.242/source/scripts/Makefile.build:503: /build/source/src/tuxedo_nb05] Error 2
make[3]: *** [/nix/store/qsfd9ca611zxw577iyabjchmdn845b79-linux-5.10.242-dev/lib/modules/5.10.242/source/scripts/Makefile.build:503: /build/source/src] Error 2
make[2]: *** [/nix/store/qsfd9ca611zxw577iyabjchmdn845b79-linux-5.10.242-dev/lib/modules/5.10.242/source/Makefile:1851: /build/source] Error 2
make[1]: *** [/nix/store/qsfd9ca611zxw577iyabjchmdn845b79-linux-5.10.242-dev/lib/modules/5.10.242/source/Makefile:192: __sub-make] Error 2
make[1]: Leaving directory &#x27;/nix/store/qsfd9ca611zxw577iyabjchmdn845b79-linux-5.10.242-dev/lib/modules/5.10.242/build&#x27;
make: *** [Makefile:29: all] Error 2</pre>
</details>
<details>
<summary>linuxKernel.packages.linux_5_15.tuxedo-drivers</summary>
<pre>      |         ^~~
/build/source/src/tuxedo_nb05/tuxedo_nb05_power_profiles.c: In function &#x27;tuxedo_nb05_power_profiles_probe&#x27;:
/build/source/src/tuxedo_nb05/tuxedo_nb05_power_profiles.c:331:9: warning: ISO C90 forbids mixed declarations and code [8;;https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wdeclaration-after-statement-Wdeclaration-after-statement8;;]
  331 |         const struct platform_device_info pinfo = {
      |         ^~~~~
/build/source/src/tuxedo_nb05/tuxedo_nb05_power_profiles.c: In function &#x27;tuxedo_nb05_power_profiles_remove&#x27;:
/build/source/src/tuxedo_nb05/tuxedo_nb05_power_profiles.c:360:9: error: implicit declaration of function &#x27;timer_delete&#x27;; did you mean &#x27;timer_reduce&#x27;? [8;;https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wimplicit-function-declaration-Werror=implicit-function-declaration8;;]
  360 |         timer_delete(&amp;profile_changed_timer);
      |         ^~~~~~~~~~~~
      |         timer_reduce
/build/source/src/tuxedo_nb05/tuxedo_nb05_power_profiles.c:361:9: warning: ISO C90 forbids mixed declarations and code [8;;https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wdeclaration-after-statement-Wdeclaration-after-statement8;;]
  361 |         struct driver_data_t *driver_data = dev_get_drvdata(&amp;wdev-&gt;dev);
      |         ^~~~~~
cc1: some warnings being treated as errors
make[4]: *** [/nix/store/m3x2ah2mf4hhk9k6cm2a26ngdknsc2y2-linux-5.15.191-dev/lib/modules/5.15.191/source/scripts/Makefile.build:289: /build/source/src/tuxedo_nb05/tuxedo_nb05_power_profiles.o] Error 1
make[3]: *** [/nix/store/m3x2ah2mf4hhk9k6cm2a26ngdknsc2y2-linux-5.15.191-dev/lib/modules/5.15.191/source/scripts/Makefile.build:552: /build/source/src/tuxedo_nb05] Error 2
make[2]: *** [/nix/store/m3x2ah2mf4hhk9k6cm2a26ngdknsc2y2-linux-5.15.191-dev/lib/modules/5.15.191/source/scripts/Makefile.build:552: /build/source/src] Error 2
make[1]: *** [/nix/store/m3x2ah2mf4hhk9k6cm2a26ngdknsc2y2-linux-5.15.191-dev/lib/modules/5.15.191/source/Makefile:1926: /build/source] Error 2
make[1]: Leaving directory &#x27;/nix/store/m3x2ah2mf4hhk9k6cm2a26ngdknsc2y2-linux-5.15.191-dev/lib/modules/5.15.191/build&#x27;
make: *** [Makefile:29: all] Error 2</pre>
</details>
</details>

--- a/pkgs/os-specific/linux/tuxedo-drivers/default.nix
+++ b/pkgs/os-specific/linux/tuxedo-drivers/default.nix
@@ -23,6 +23,11 @@ stdenv.mkDerivation (finalAttrs: {
 
   patches = [ ./no-cp-etc-usr.patch ];
 
+  postInstall = ''
+    echo "Running postInstallhook"
+    install -Dm 0644 -t $out/etc/udev/rules.d usr/lib/udev/rules.d/*
+  '';
+
   buildInputs = [ pahole ];
   nativeBuildInputs = [ kmod ] ++ kernel.moduleBuildDependencies;
 

--- a/pkgs/os-specific/linux/tuxedo-drivers/default.nix
+++ b/pkgs/os-specific/linux/tuxedo-drivers/default.nix
@@ -7,6 +7,7 @@
   kmod,
   pahole,
   gitUpdater,
+  udevCheckHook,
 }:
 
 stdenv.mkDerivation (finalAttrs: {
@@ -29,13 +30,19 @@ stdenv.mkDerivation (finalAttrs: {
   '';
 
   buildInputs = [ pahole ];
-  nativeBuildInputs = [ kmod ] ++ kernel.moduleBuildDependencies;
+  nativeBuildInputs = [
+    kmod
+    udevCheckHook
+  ]
+  ++ kernel.moduleBuildDependencies;
 
   makeFlags = kernelModuleMakeFlags ++ [
     "KERNELRELEASE=${kernel.modDirVersion}"
     "KDIR=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build"
     "INSTALL_MOD_PATH=${placeholder "out"}"
   ];
+
+  doInstallCheck = true;
 
   passthru.updateScript = gitUpdater {
     rev-prefix = "v";

--- a/pkgs/os-specific/linux/tuxedo-drivers/default.nix
+++ b/pkgs/os-specific/linux/tuxedo-drivers/default.nix
@@ -11,15 +11,17 @@
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "tuxedo-drivers-${kernel.version}";
-  version = "4.14.0";
+  version = "4.15.4";
 
   src = fetchFromGitLab {
     group = "tuxedocomputers";
     owner = "development/packages";
     repo = "tuxedo-drivers";
     rev = "v${finalAttrs.version}";
-    hash = "sha256-79YZaK8WrHOxSUJWxi4lc+foh4xz3EgRnjw+OrL8yqU=";
+    hash = "sha256-WJeju+czbCw03ALW7yzGAFENCEAvDdKqHvedchd7NVY=";
   };
+
+  patches = [ ./no-cp-etc-usr.patch ];
 
   buildInputs = [ pahole ];
   nativeBuildInputs = [ kmod ] ++ kernel.moduleBuildDependencies;

--- a/pkgs/os-specific/linux/tuxedo-drivers/no-cp-etc-usr.patch
+++ b/pkgs/os-specific/linux/tuxedo-drivers/no-cp-etc-usr.patch
@@ -1,0 +1,12 @@
+diff --git a/Makefile b/Makefile
+index 75a6bc1..6021d42 100644
+--- a/Makefile
++++ b/Makefile
+@@ -30,7 +30,6 @@ all:
+ 
+ install: all
+ 	make -C $(KDIR) M=$(PWD) $(MAKEFLAGS) modules_install
+-	cp -r etc usr /
+ 
+ clean:
+ 	make -C $(KDIR) M=$(PWD) $(MAKEFLAGS) clean

